oxenss/storage/database.cpp

```diff
@@ -287,6 +287,30 @@ ALTER TABLE messages ADD COLUMN namespace INTEGER NOT NULL DEFAULT 0;
             )");
         }
 
+        if (!db.tableExists("revoked_subkeys")) {
+            log::info(logcat, "Upgrading database schema");
+            db.exec(R"(
+CREATE TABLE revoked_subkeys (
+    id INTEGER,
+    owner INTEGER REFERENCES owners(id) ON DELETE CASCADE,
+    subkey TEXT NOT NULL,
+    timestamp INTEGER NOT NULL DEFAULT CURRENT_TIMESTAMP,
+
+    UNIQUE(subkey)
+);
+
+CREATE TRIGGER IF NOT EXISTS subkey_autoclean
+    AFTER INSERT ON revoked_subkeys WHEN (select count(*) FROMrevoked_subkeys WHERE owner = NEW.owner) > 50
+    BEGIN
+        DELETE FROM revoked_subkeys
+            WHERE owner = NEW.owner and subkey NOT IN (SELECT subkey
+                                                          FROM revoked_subkeys WHERE owner = NEW.owner
+                                                          ORDER BY timestamp DESC LIMIT 50
```

Comment:
That indent seems a bit excessive, perhaps:
```suggestion
            WHERE owner = NEW.owner AND subkey NOT IN (
                SELECT subkey FROM revoked_subkeys
                WHERE owner = NEW.owner
                ORDER BY timestamp DESC LIMIT 50
```